graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java

```diff
@@ -124,21 +134,27 @@ public GrokPattern updatePattern(@ApiParam(name = "patternId", required = true)
                                      @ApiParam(name = "pattern", required = true)
                                      GrokPatternSummary pattern) throws NotFoundException, ValidationException {
         checkPermission(RestPermissions.INPUTS_EDIT);
-        
+
         final GrokPattern oldPattern = grokPatternService.load(patternId);
-        
+
+        final Set<String> updatedNames = Sets.newHashSet(pattern.name, oldPattern.name);
```

Comment:
Is this really the semantics of the `GrokPatternsChangedEvent#updatedPatterns` field? Shouldn't a renamed pattern produce an event which contains the old name in `GrokPatternsChangedEvent#deletedPatterns` (because the pattern with that doesn't exist anymore)?